### DISCUSSION TITLE
use flutter code with flet code.

### DISCUSSION THREAD

**Discussion by hahelui** [flet-dev/flet#3343] [Ideas]
This might already exist, but I didn't find it. Flet is very cool, awesome, and easy to use. However, it still lacks some functionalities and widgets. Some widgets are broken now, like the Autocomplete widget, for example. Wouldn't it be great to have a way to create a component or widget in Flutter code and use it inside Flet code?

**Comment by FeodorFitsner** [comment]
That's already possible!

There is docs in progress: https://github.com/flet-dev/website/blob/extending-flet/docs/extend/integrating-existing-flutter-packages.md - you are welcome to check/try/ask.

Sample Flutter packages: https://github.com/flet-dev/flet/tree/main/packages - see those `flet_*` packages for example. And you can easily find their Python counterparts here: https://github.com/flet-dev/flet/tree/main/sdk/python/packages/flet-core/src/flet_core

**Reply by hahelui** [reply]
Wow, That is awesome, I'll check it out. Thank you for the hard work.
